### PR TITLE
documents: display all fields on the detail view

### PR DESCRIPTION
**Pull request by Garfield-fr**
Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Why are you opening this PR?

- Add some fields.

## How to test?

- Add data to some fields and check the display on "Header" and "Description" tab

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by iGormilhit — CHANGES_REQUESTED** [review]
Commit message title proposition: `documents: display all fields on the detail view`

**Review by iGormilhit — APPROVED** [review]

File: projects/admin/src/app/record/detail-view/document-detail-view/document-description/document-description.component.ts [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    }
+    return cartographic;
+  }
+
+  /** Get list of document edition statement */
+  get editionStatement(): any[] {
+    const results = [];
+    if ('seriesStatement' in this.record.metadata) {
+      const statements = this.record.metadata.seriesStatement;
+      statements.forEach((element: any) => {
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-description/document-description.component.ts:56]: cosmetic : could be merge
```javascript
this.record.metadata.seriesStatement.forEach((element: any) => { ...});
```

File: projects/admin/src/app/record/detail-view/document-detail-view/document-description/document-description.component.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    }
+    return results;
+  }
+
+  /** Get list of document identifiers */
+  get identifiedBy(): any[] {
+    const identifiers = [];
+    if ('identifiedBy' in this.record.metadata) {
+      const identified = this.record.metadata.identifiedBy;
+      identified.forEach((id: any) => {
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-description/document-description.component.ts:80]: same

File: projects/admin/src/app/record/detail-view/document-detail-view/document-description/document-description.component.ts [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+        const details = [];
+        // Replace bf:Local by source
+        let idType = id.type;
+        if (id.type === 'bf:Local') {
+          idType = id.source;
+        }
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-description/document-description.component.ts:86]:
```javacsript 
const idtype = (id.type === 'bf:Local') ? id.source : id.type;
```

File: projects/admin/src/app/record/detail-view/document-detail-view/document-description/document-description.component.ts [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+  get provisionActivity() {
+    const provisions = [];
+    if ('provisionActivity' in this.record.metadata) {
+      this.record.metadata.provisionActivity.forEach((provision: any) => {
+        if ('original_date' in provision) {
+          provisions.push(provision.original_date);
+        }
+      });
+    }
+    return provisions;
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-description/document-description.component.ts:137]:
```typecsript
return ('provisionActivity' in this.record.metadata)
  ? this.record.metadata.provisionActivity.filter(provision => 'original_date' in provision)
  : []
```

File: projects/admin/src/app/scss/styles.scss [hunk 5]
```diff
@@ -275,3 +275,24 @@ json-schema-form {
   background-color: #f8f9fa;
   border-radius: 5px;
 }
+
+.div-striped {
+  padding: 0;
+}
```
Comment (zannkukai) [inline projects/admin/src/app/scss/styles.scss:281]: could you use a `p-0` on the HTML element instead.

File: projects/admin/src/app/scss/styles.scss [hunk 6]
```diff
@@ -275,3 +275,24 @@ json-schema-form {
...
+  padding-top: 2px;
+  padding-bottom: 2px;
+}
+
+.div-striped .striped:nth-of-type(odd) {
+  background: $light;
+}
+
+.div-striped .striped:nth-of-type(even) {
+  background: #FFFFFF;
```
Comment (zannkukai) [inline projects/admin/src/app/scss/styles.scss:293]: `transparent` instead of `white` ?

File: projects/admin/src/app/scss/styles.scss [hunk 7]
```diff
@@ -275,3 +275,24 @@ json-schema-form {
...
+}
+
+.div-striped .striped .row div:first-of-type  {
+  padding-left: 20px;
```
Comment (zannkukai) [inline projects/admin/src/app/scss/styles.scss:297]: ```map-get($spacers, 1);``` (1, 2, 3, ... correspond to p-1, p-2, p-3, ...)

**Review by zannkukai — APPROVED** [review]

File: projects/admin/src/app/record/detail-view/document-detail-view/document-description/document-description.component.ts [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+      });
+    }
+    return variants;
+  }
+
+  /**
+   * Constructor
+   * @param translateService - TranslateService
+   */
+  constructor(public translateService: TranslateService) {}
```
Comment (sebastiendeleze) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-description/document-description.component.ts:150]: The constructor should be before the getters

**Review by sebastiendeleze — APPROVED** [review]

**Review by benerken — APPROVED** [review]

**Pull request merged**